packages/arb-token-bridge-ui/src/hooks/useNetworksAndSigners.tsx

```diff
@@ -227,62 +244,85 @@ export function NetworksAndSignersProvider(
       return
     }
 
-    // Case 3
-    getL1Network(provider as Web3Provider)
-      .then(async l1Network => {
-        // Web3Provider is connected to an L1 network. We instantiate a provider for the L2 network.
-        const l2Provider = new StaticJsonRpcProvider(
+    // Case 4
+    getParentChain(provider as Web3Provider)
```

Comment:
`ParentChain` is either an L1 network or an L2 network that is a parent to at least one Orbit chain, specified in its property `partnerChainIDs`.